sentry-options-validation/src/lib.rs

```diff
@@ -303,6 +311,154 @@ impl Default for SchemaRegistry {
     }
 }
 
+/// Watches the values directory for changes, reloading if there are any.
+///
+/// Does not do an initial fetch, assumes the caller has already loaded values.
+/// Child thread may panic if we run out of memory or cannot create more threads.
+///
+/// Uses polling for now, could use `inotify` or similar later on.
+pub struct ValuesWatcher {
+    stop_signal: Arc<AtomicBool>,
+    thread: Option<JoinHandle<()>>,
+}
+
+impl ValuesWatcher {
+    /// Creates a new ValuesWatcher struct and spins up the watcher thread
+    ///
+    /// Returns a FileRead error if we fail to get metadata for the directory
+    pub fn new(
+        values_path: &Path,
+        registry: Arc<SchemaRegistry>,
+        values: Arc<RwLock<ValuesByNamespace>>,
+    ) -> ValidationResult<Self> {
+        // validate permissions and existence of directory
+        fs::metadata(values_path)?;
+
+        let stop_signal = Arc::new(AtomicBool::new(false));
```

Comment:
Should we attempt to stop people from creating multiple file watcher threads here? Can be temporary but in the case folks decide to create multiple instances of the client

Reply:
Perhaps some flag to know if watcher is alive or not would help